packages/frontend-v2/utils/plan/updatePlanOnDragEnd.ts

```diff
@@ -29,82 +29,95 @@ import { isCourseInTerm } from "./isCourseInTerm";
     const scheduleTerms = flattenScheduleToTerms<string>(draftPlan.schedule);
 
-    const newTerm = scheduleTerms.find(
-      (term) => term.id === draggedOverTerm.id
-    );
-
-    if (!newTerm) {
-      throw new Error("Term the course is dragged over isn't found");
-    }
+    // If it is not being dragged over a droppable, then we delete the course
+    if (draggedOverTerm == null) {
+      const newTerm = scheduleTerms.find(term => term.classes.some((course) => course.id === draggedCourse.id));
+      if (!newTerm) {
+        throw new Error("Term the course is dragged from isn't found");
+      }
 
-    if (!draggedCourse.data.current?.course) {
-      throw new Error(
-        "The course being dragged is not found in the term it is being dragged from"
+      // remove the class from the old term
+      newTerm.classes = newTerm.classes.filter(
+        (course) => course.id !== draggedCourse.id
       );
-    }
-
-    const draggedCourseDetails: ScheduleCourse2<unknown> =
-      draggedCourse.data.current.course;
-
-    const oldTerm = scheduleTerms.find((term) =>
-      term.classes.some((course) => course.id === draggedCourse.id)
-    );
-
-    const isFromSidebar = draggedCourse.data.current.isFromSidebar;
-    const isSameTerm = !isFromSidebar && oldTerm && oldTerm.id === newTerm.id;
-
-    /*
-     * Prevent duplicate courses in the same term,
-     * don't need to display error if course is not changing terms.
-     */
-    if (
-      isCourseInTerm(
-        draggedCourseDetails.classId,
-        draggedCourseDetails.subject,
-        newTerm
-      ) &&
-      !isSameTerm
-    ) {
-      toast.error(
-        `Oops, ${getCourseDisplayString(
-          draggedCourseDetails
-        )} already exists in Year ${newTerm.year}, ${getSeasonDisplayWord(
-          newTerm.season
-        )}.`
+    } else {
+      const newTerm = scheduleTerms.find(
+        (term) => term.id === draggedOverTerm.id
       );
-      throw new Error("Duplicate course in term.");
-    }
 
-    /*
-     * Course is from a term, so we need to move it, we don't need to move
-     * courses that are from the sidebar.
-     */
-    if (!isFromSidebar) {
-      if (!oldTerm) {
-        throw new Error("Term the course is dragged from isn't found");
+      if (!newTerm) {
+        throw new Error("Term the course is dragged over isn't found");
       }
 
-      if (isSameTerm) {
-        throw new Error("Course is being dragged over its own term");
+      if (!draggedCourse.data.current?.course) {
+        throw new Error(
+          "The course being dragged is not found in the term it is being dragged from"
+        );
       }
 
-      // remove the class from the old term and add it to the new term
-      oldTerm.classes = oldTerm.classes.filter(
-        (course) => course.id !== draggedCourse.id
+      const draggedCourseDetails: ScheduleCourse2<unknown> =
+        draggedCourse.data.current.course;
+
+      const oldTerm = scheduleTerms.find((term) =>
```

Comment:
this bit looks like the code at my above comment-- i think u can just move it to the top to avoid recalculating the old term, but if it messes up the logic then j ignore